### ISSUE TITLE
Output and ground truth summary may be misplaced

### ISSUE THREAD

**Issue by nlpmc** [nlpyang/hiersumm#2]
When I turned the mode to "test" and changed the dataset to test dataset, I found that the events described in the generated summary did not match the ground summary. The summary just describes another thing or event which is totally different from the ground truth summary. 
I try to debug this code:
the input batch of data is 
![batch](https://user-images.githubusercontent.com/52098091/59961427-bfa38900-94e8-11e9-929e-573445d3b4cd.png)
As you can see in this screenshot, the 5th data is about CCGS and the 6th data is about Lu yan yuan lu.
But in the translated text:
![translate](https://user-images.githubusercontent.com/52098091/59961442-ee216400-94e8-11e9-81ed-e32be3d96090.png)
The 2nd data's ground truth is about CCGS, but the summary is about the royal air force.
In the 4th case, the generated summary also describes the royal air force, but the ground truth describes the lu yan yuan lu.

I don't know if this is a bug in the code or if there are some mistakes when I tested the model.
I'm looking forward to your reply. Thank you!

**Comment by nlpyang** [comment]
please paste your command, or I can't see what's happening here

**Comment by gsh199449** [comment]
My command is 
``` bash
python3 train_abstractive.py -data_path /data/dataset/multi-doc-wiki/WIKI -dataset test -mode test -batch_size 10500 -seed 666 -train_steps 1000000 -save_checkpoint_steps 5000 -report_every 100 -trunc_tgt_ntoken 400 -trunc_src_nblock 24 -visible_gpus 0 -gpu_ranks 0 -world_size 1 -accum_count 1 -dec_dropout 0.1 -enc_dropout 0.1 -label_smoothing 0.1 -vocab_path /data/dataset/multi-doc-wiki/spm9998_3.model -model_path /data/logs/hiersumm -test_from /data/logs/hiersumm/model_step_350000.pt -log_file /data/logs/hiersumm/logs.txt -inter_layers 6,7 -inter_heads 8 -hier
```

**Comment by nlpmc** [comment]
The command is 
```
python3 train_abstractive.py -data_path /data/dataset/multi-doc-wiki/WIKI -dataset test -mode test -batch_size 10500 -seed 666 -train_steps 1000000 -save_checkpoint_steps 5000 -report_every 100 -trunc_tgt_ntoken 400 -trunc_src_nblock 24 -visible_gpus 0 -gpu_ranks 0 -world_size 1 -accum_count 1 -dec_dropout 0.1 -enc_dropout 0.1 -label_smoothing 0.1 -vocab_path /data/dataset/multi-doc-wiki/spm9998_3.model -model_path /data/logs/hiersumm -test_from /data/logs/hiersumm/model_step_350000.pt -log_file /data/logs/hiersumm/logs.txt -inter_layers 6,7 -inter_heads 8 -hier
```
I only have one GPU, so I change the gpu_ranks and world_size.

**Comment by nlpyang** [comment]
I do find a bug in generating the raw_src files. Please pull the repo, this should be solved now.

Since raw_src files are only used for analysis, the results will not be influenced.

Thanks for pointing this out.

**Comment by nlpmc** [comment]
Thanks for your quick reply! Your recent patch works well, it fixes a part of the problem. The ground truth summary and source document are matched. 

But the generated summary may be still misplaced?
![aa](https://user-images.githubusercontent.com/52098091/59966787-239e6f80-9532-11e9-8dcf-1665da468198.png)
As you can see in this screenshot, the 6th summary describes "royal air force" but the source document and ground summary describe the lu yan yuan lu. The 5th summary also has this problem.

This decode result is the 350000 step checkpoint, the sentence generated by this checkpoint is fluent. But some sentences describe the same event.
generated summary file
![result](https://user-images.githubusercontent.com/52098091/59966844-f8685000-9532-11e9-994e-0fe950e85fd6.png)

Ground Truth File:
![gold](https://user-images.githubusercontent.com/52098091/59966849-fef6c780-9532-11e9-860b-3cb3a631e852.png)

**Comment by nlpyang** [comment]
This is not a bug, this happens in all language generation tasks when your model is not efficiently trained. The model will remember the training set and generate fluent but non-related content.

**Comment by nlpmc** [comment]
As you said in the paper, your model has trained 500,000 steps, and my current model has only trained 350,000 steps. I will continue to train it.
Thanks for your patience!

**Comment by nlpyang** [comment]
The batch_size is also important. If you only have on gpu, you need to have a much larger batch size or you could increase the steps from gradient accumulation.

**Comment by nlpmc** [comment]
OK, I will have a try. Thanks for your advice!
